### PR TITLE
Feat/update angular 14

### PR DESCRIPTION
**Pull request by Sumragen**

### CODE REVIEW

**Review by grankovsky — APPROVED** [review]

**Review by sergeysenja1992 — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,37 @@
+
+
```
Comment (Kaskyi) [inline CHANGELOG.md:2]: add description like 
```
# 3.2.3 (2021-05-27)

### Features
```

File: tsconfig.json [hunk 2]
```diff
@@ -57,8 +58,10 @@
       "@xm-ngx/dashboard": ["packages/dashboard/src"],
       "@xm-ngx/dashboard/testing": ["packages/dashboard/testing"],
       "@xm-ngx/xm-balance": ["src/app/xm-balance"],
-      "@xm-ngx/components/language": ["src/app/shared/language"]
-
+      "@xm-ngx/components/language": ["src/app/shared/language"],
+      "rxjs": ["node_modules/rxjs"],
```
Comment (Kaskyi) [inline tsconfig.json:62]: not sure if this is necessary

File: src/styles.scss [hunk 3]
```diff
@@ -1,39 +1,46 @@
-/* You can add global styles to this file, and also import other style files */
+/* WARNING: Override of global ./src/styles.scss */
 
 // Fonts
 $material-icons-font-path: '~assets/fonts/material-design-icons-v5.woff2';
 @import '~@xm-ngx/styles/src/material-design-icons';
 
 // Theme
-@import './styles/theme';
```
Comment (Kaskyi) [inline src/styles.scss:8]: vodafone

**Review by Kaskyi — CHANGES_REQUESTED** [review]

File: src/app/modules/xm-shared/xm-shared.module.ts [hunk 4]
```diff
@@ -27,7 +26,7 @@ import { MatModule } from '../../mat.module';
         XmAlertModule,
         FormsModule,
         ReactiveFormsModule,
-        XmAlertModule,
+        // XmAlertModule,
```
Comment (Kaskyi) [inline src/app/modules/xm-shared/xm-shared.module.ts:29]: remove?

File: proxy.conf.js [hunk 5]
```diff
@@ -7,7 +7,8 @@ const PROXY_CONFIG = [
         pathRewrite: {
             [PROXY_SERVER_URL_REG]: '',
         },
-        target: "http://xm.test.xm-online.com.ua",
+        // target: "http://xm.test.xm-online.com.ua",
```
Comment (Kaskyi) [inline proxy.conf.js:10]: vf

File: packages/components/sidebar-right/xm-sidebar-right.module.ts [hunk 6]
```diff
@@ -71,7 +71,10 @@ export class XmSidebarRight implements OnInit, OnDestroy {
         const viewContainerRef = this.xmContainerOutlet.viewContainerRef;
         const componentFactoryResolver = config.injector.get(ComponentFactoryResolver);
         const cfr = componentFactoryResolver.resolveComponentFactory(templateRef);
-        const c = viewContainerRef.createComponent<T>(cfr, null, config.injector);
+        const c = viewContainerRef.createComponent<T>(cfr.componentType, {
+            injector: config.injector,
+            ngModuleRef: cfr['ngModule']
```
Comment (Kaskyi) [inline packages/components/sidebar-right/xm-sidebar-right.module.ts:76]: should be an variable

File: packages/alert/src/xm-alert.service.ts [hunk 7]
```diff
@@ -36,22 +35,24 @@ export class XmAlertService {
...
             confirmButtonText: this.config.yesLabel,
             cancelButtonText: this.config.cancelLabel,
         };
         settings = _.merge(DEFAULT, settings);
 
         if (settings.title) {
             const opts = settings.titleOptions || {};
-
-            settings.title = this.xmTranslateService.translate(settings.title, opts);
+            // TODO: Check settings.title type
```
Comment (Kaskyi) [inline packages/alert/src/xm-alert.service.ts:49]: remove?

File: packages/alert/src/toaster/alert-error.component.ts [hunk 8]
```diff
@@ -230,8 +232,9 @@ export class JhiAlertErrorComponent implements OnDestroy {
             this.showError(res.error.error);
         } else if (res.error?.detail) {
             this.showError(res.error.detail);
-        } else if (res.title) {
-            this.showError(res.title);
+            // TODO: handle type incompatibility
```
Comment (Kaskyi) [inline packages/alert/src/toaster/alert-error.component.ts:235]: remove?

**Review by oscaryald — DISMISSED** [review]

File: tsconfig.json (hunk 2 above)
Comment (Sumragen) [reply tsconfig.json:62]: otherwise we will retrieve ts errors due to incompatible versions of nested dependencies

File: CHANGELOG.md (hunk 1 above)
Comment (Sumragen) [reply CHANGELOG.md:2]: Nice point. I'll fill detailed info about changes

File: src/styles.scss (hunk 3 above)
Comment (Sumragen) [reply src/styles.scss]: reverted that miss

File: src/styles/_theme.scss [hunk 9]
```diff
@@ -1,81 +1,165 @@
-/*
- * WARNING: This file can be overridden by your custom theme,
- *          if you use the following command in your development:
- *             node ./scripts/replace-styles.js theme your-theme.scss
- */
+// Theme Styles
```
Comment (EKrasovskiy) [inline src/styles/_theme.scss:1]: shoul be reverted

File: packages/styles/src/vendor/_chartist.scss [hunk 10]
```diff
@@ -20,7 +21,7 @@ $ct-series-colors: (
 
 $ct-label-color:   rgba(0, 0, 0, 0.4) !default;
 
-@import "~chartist/dist/scss/chartist";
+//@import "~chartist/dist/index.css";
```
[the remark this answers is not in the document]
Comment (Sumragen) [reply packages/styles/src/vendor/_chartist.scss:24]: Not removed, but moved above

File: src/styles/_theme.scss (hunk 9 above)
Comment (Sumragen) [reply src/styles/_theme.scss]: agree

File: packages/alert/src/xm-alert.service.ts (hunk 7 above)
Comment (Sumragen) [reply packages/alert/src/xm-alert.service.ts:49]: isn't removed. it's below

**Comment by Kaskyi** [comment]
fixes https://github.com/xm-online/xm-webapp/issues/1183#issue-961704145

File: packages/components/sidebar-right/xm-sidebar-right.module.ts (hunk 6 above)
Comment (Sumragen) [reply packages/components/sidebar-right/xm-sidebar-right.module.ts:100]: This `hack` will be changed in next PR dedicated to dynamic loaders refactoring

File: src/app/modules/xm-shared/xm-shared.module.ts (hunk 4 above)
Comment (Sumragen) [reply src/app/modules/xm-shared/xm-shared.module.ts:29]: it's a duplicate of 26's line

**Review by Kaskyi — APPROVED** [review]

**Review by Olensavchenko — APPROVED** [review]

**Review by LevBD — APPROVED** [review]

**Pull request merged**